sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/ier12.processing.service.ts

```diff
@@ -301,12 +318,235 @@ export class IER12ProcessingService {
         totalAssessedCost: assessmentData.totalAssessedCost,
         totalAssessmentNeed: assessmentData.totalAssessmentNeed,
         disbursementSentDate: disbursement.dateSent,
+        applicationEventCode: await this.getApplicationEventCode(
+          pendingAssessment,
+          disbursement,
+        ),
       };
       ier12Records.push(ier12Record);
     }
     return ier12Records;
   }
 
+  /**
+   * Get application event code (i.e current state of an application)
+   * @param assessment assessment of institutions.
+   * @param currentDisbursementSchedule current disbursement schedule.
+   * @returns application event code.
+   */
+  private async getApplicationEventCode(
+    assessment: StudentAssessment,
```

Comment:
I believe, we can just pass the applicationNumber and applicationStatus instead of the entire assessment object.

I see that studentId is also used from application to be used in the restriction method, but if we re-use `checkActiveRestriction ` as per the other comment we do not need it.

Reply:
For the parent method, I didn't want to pick and send the variable, But if devs find it not a good idea I can change it.
Regarding the student_id, Pls let me know your reply on the comment https://github.com/bcgov/SIMS/pull/2374#discussion_r1353258233

Reply:
Thanks for making changes as per our discussion.